File: pglite/src/pglite/release/fs/fsBridge.ts
import type { FsPathEntry, RuntimeModule } from "../core/runtimeTypes.js";

export function applyFsBootstrapPaths(
	module: RuntimeModule,
	paths: FsPathEntry[],
): void {
	const createPath = module.FS_createPath;
	if (!createPath) {
		throw new Error("FS_createPath is not available on module");
	}
	for (const entry of paths) {
		createPath(entry.parent, entry.name, entry.canRead, entry.canWrite);
	}
}

File: src/main/java/io/github/hidekatsu_izuno/pglite_jdbc/pglite/release/fs/fsBridge.java
package io.github.hidekatsu_izuno.pglite_jdbc.pglite.release.fs;

import io.github.hidekatsu_izuno.pglite_jdbc.pglite.release.core.runtimeTypes;
import java.util.List;

public class fsBridge {
    private fsBridge() {}

    public static void applyFsBootstrapPaths(
        runtimeTypes.RuntimeModule module,
        List<runtimeTypes.FsPathEntry> paths
    ) {
        if (module == null) {
            throw new IllegalArgumentException("module is null");
        }
        for (var entry : paths) {
            module.FS_createPath(entry.parent(), entry.name(), entry.canRead(), entry.canWrite());
        }
    }
}
